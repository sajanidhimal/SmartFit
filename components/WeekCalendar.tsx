import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

interface DayItem {
  day: string;
  date: number;
  active: boolean;
}

interface WeekCalendarProps {
  weekDays: DayItem[];
  onPrevWeek: () => void;
  onNextWeek: () => void;
  onSelectDay: (index: number) => void;
}

export default function WeekCalendar({
  weekDays,
  onPrevWeek,
  onNextWeek,
  onSelectDay
}: WeekCalendarProps) {
  return (
    <View className="flex-row items-center justify-between py-4 bg-white rounded-lg my-4">
      <TouchableOpacity className="px-2" onPress={onPrevWeek}>
        <Ionicons name="chevron-back" size={24} color="#ccc" />
      </TouchableOpacity>
      
      <View className="flex-row flex-1 justify-around">
        {weekDays.map((item, index) => (
          <TouchableOpacity 
            key={index} 
            className="items-center"
            onPress={() => onSelectDay(index)}
          >
            <Text className="text-gray-500 text-xs mb-1">{item.day}</Text>
            <View className={`w-8 h-8 rounded-full items-center justify-center ${item.active ? 'bg-secondary' : 'bg-gray-100'}`}>
              <Text className={`${item.active ? 'text-white' : 'text-gray-500'}`}>
                {item.date}
              </Text>
            </View>
          </TouchableOpacity>
        ))}
      </View>
      
      <TouchableOpacity className="px-2" onPress={onNextWeek}>
        <Ionicons name="chevron-forward" size={24} color="#ccc" />
      </TouchableOpacity>
    </View>
  );
}