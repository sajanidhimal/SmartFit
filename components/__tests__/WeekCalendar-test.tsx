import React from 'react';
import renderer, { act } from 'react-test-renderer';
import WeekCalendar from '../WeekCalendar';

describe('WeekCalendar', () => {
  const mockWeekDays = [
    { day: 'Mon', date: 1, active: false },
    { day: 'Tue', date: 2, active: true },
    { day: 'Wed', date: 3, active: false },
    { day: 'Thu', date: 4, active: false },
    { day: 'Fri', date: 5, active: false },
    { day: 'Sat', date: 6, active: false },
    { day: 'Sun', date: 7, active: false },
  ];
  
  const mockOnPrevWeek = jest.fn();
  const mockOnNextWeek = jest.fn();
  const mockOnSelectDay = jest.fn();
  
  beforeEach(() => {
    // Clear mocks before each test
    mockOnPrevWeek.mockClear();
    mockOnNextWeek.mockClear();
    mockOnSelectDay.mockClear();
  });
  
  it('renders correctly with provided days', () => {
    const tree = renderer.create(
      <WeekCalendar 
        weekDays={mockWeekDays}
        onPrevWeek={mockOnPrevWeek}
        onNextWeek={mockOnNextWeek}
        onSelectDay={mockOnSelectDay}
      />
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });
  
  it('calls onPrevWeek when the left chevron is pressed', () => {
    const component = renderer.create(
      <WeekCalendar 
        weekDays={mockWeekDays}
        onPrevWeek={mockOnPrevWeek}
        onNextWeek={mockOnNextWeek}
        onSelectDay={mockOnSelectDay}
      />
    );
    
    const prevButton = component.root.findAllByProps({ name: 'chevron-back' })[0]
      .parent.parent;
      
    act(() => {
      prevButton.props.onPress();
    });
    
    expect(mockOnPrevWeek).toHaveBeenCalledTimes(1);
  });
  
  it('calls onNextWeek when the right chevron is pressed', () => {
    const component = renderer.create(
      <WeekCalendar 
        weekDays={mockWeekDays}
        onPrevWeek={mockOnPrevWeek}
        onNextWeek={mockOnNextWeek}
        onSelectDay={mockOnSelectDay}
      />
    );
    
    const nextButton = component.root.findAllByProps({ name: 'chevron-forward' })[0]
      .parent.parent;
      
    act(() => {
      nextButton.props.onPress();
    });
    
    expect(mockOnNextWeek).toHaveBeenCalledTimes(1);
  });
  
  it('calls onSelectDay with the correct index when a day is pressed', () => {
    const component = renderer.create(
      <WeekCalendar 
        weekDays={mockWeekDays}
        onPrevWeek={mockOnPrevWeek}
        onNextWeek={mockOnNextWeek}
        onSelectDay={mockOnSelectDay}
      />
    );
    
    // Find all day touchables and press the third one (index 2)
    const dayButtons = component.root.findAllByProps({ 
      className: "items-center"
    });
    
    act(() => {
      dayButtons[2].props.onPress();
    });
    
    expect(mockOnSelectDay).toHaveBeenCalledWith(2);
  });
}); 