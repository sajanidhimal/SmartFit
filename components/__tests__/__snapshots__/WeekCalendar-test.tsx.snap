// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekCalendar renders correctly with provided days 1`] = `
<View
  className="flex-row items-center justify-between py-4 bg-white rounded-lg my-4"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      color="#ccc"
      name="chevron-back"
      size={24}
    />
  </View>
  <View
    className="flex-row flex-1 justify-around"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        className="text-gray-500 text-xs mb-1"
      >
        Mon
      </Text>
      <View
        className="w-8 h-8 rounded-full items-center justify-center bg-gray-100"
      >
        <Text
          className="text-gray-500"
        >
          1
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        className="text-gray-500 text-xs mb-1"
      >
        Tue
      </Text>
      <View
        className="w-8 h-8 rounded-full items-center justify-center bg-secondary"
      >
        <Text
          className="text-white"
        >
          2
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        className="text-gray-500 text-xs mb-1"
      >
        Wed
      </Text>
      <View
        className="w-8 h-8 rounded-full items-center justify-center bg-gray-100"
      >
        <Text
          className="text-gray-500"
        >
          3
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        className="text-gray-500 text-xs mb-1"
      >
        Thu
      </Text>
      <View
        className="w-8 h-8 rounded-full items-center justify-center bg-gray-100"
      >
        <Text
          className="text-gray-500"
        >
          4
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        className="text-gray-500 text-xs mb-1"
      >
        Fri
      </Text>
      <View
        className="w-8 h-8 rounded-full items-center justify-center bg-gray-100"
      >
        <Text
          className="text-gray-500"
        >
          5
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        className="text-gray-500 text-xs mb-1"
      >
        Sat
      </Text>
      <View
        className="w-8 h-8 rounded-full items-center justify-center bg-gray-100"
      >
        <Text
          className="text-gray-500"
        >
          6
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        className="text-gray-500 text-xs mb-1"
      >
        Sun
      </Text>
      <View
        className="w-8 h-8 rounded-full items-center justify-center bg-gray-100"
      >
        <Text
          className="text-gray-500"
        >
          7
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      color="#ccc"
      name="chevron-forward"
      size={24}
    />
  </View>
</View>
`;
